api/src/controllers/login.js

```diff
@@ -181,7 +181,8 @@ module.exports = {
             }
             res.send(body);
           });
-        });
+        })
+        .catch(err => logger.error('Could not find branding doc on couchdb: %o', err));
```

Comment:
I think we should still render the login page even if the branding doc is not found - otherwise nobody will be able to log in!

Consider replacing the `catch` on line 170 with this:

```js
      .catch(() => {
        return db.medic.get('branding', {attachments: true})
          .then(doc => {
            const image = doc._attachments[doc.resources.logo];
            return {
              name: doc.title,
              logo: `data:${image.content_type};base64,${image.data}`
            }
          })
          .catch(err => logger.error('Could not find branding doc on couchdb: %o', err));
      })
      .then(branding => {
        renderLogin(redirect, branding, (err, body) => {
          if (err) {
            logger.error('Could not find login page');
            throw err;
          }
          res.send(body);
        });
      });
```

NB: I haven't tested this - it may not even compile! Check that this works by deleting the branding doc and rendering the login page.

It'd be good to add a unit test for this case too - it's unlikely to happen but it's important people can log in!

Reply:
Yeah those tests were bound run always, they never had `return` statement so they were "All Weather Road" Lol!